pkg/solana/logpoller/filters.go

```diff
@@ -291,7 +302,7 @@ func (fl *filters) MatchingFiltersForEncodedEvent(event ProgramEvent) iter.Seq[F
 		// discriminators if the first 10 characters don't match. If it passes that initial test, we base64-decode the
 		// first 12 characters, and use the first 8 bytes of that as the event sig to call MatchingFilters. The address
 		// also needs to be base58-decoded to pass to MatchingFilters
-		_, ok = fl.knownDiscriminators[event.Data[:10]]
+		_, ok = fl.knownDiscriminators[base64.StdEncoding.EncodeToString(discriminator)]
```

Comment:
You decode it and then immediately encode it again. Then decode it again later? There's no need to do 3 steps we only need one quick check and then one decoding if it passes.

Reply:
(Also, the comments don't match any more what the code does.)